draft-ietf-cbor-7049bis.md

```diff
@@ -408,27 +409,31 @@ floats as integers or vice versa, perhaps to save encoded bytes.
 
 A CBOR data item  ({{cbor-data-models}}) is encoded to or decoded from
 a byte string as described in this section.  The encoding is
-summarized in {{jumptable}}.
+summarized in {{jumptable}}.  An encoder MUST produce only well-formed
+items.  A decoder MUST stop and return an error with no data when it
+encounters a non-well-formed item.
 
 The initial byte of each encoded data item contains both information
 about the major type (the high-order 3 bits, described in
 {{majortypes}}) and additional information (the low-order 5 bits).
-Additional information value
-31 is used for indefinite-length items, described in {{indefinite}}.
-Additional information values 28 to 30 are reserved for future
-expansion.
-
-Additional information values from 0 to 27 describes how to construct an
-"argument", possibly consuming additional bytes.  For major type 7 and
-additional information 25 to 27 (floating
-point numbers), there is a special case; in all other cases
-the additional information value, possibly combined with following
-bytes, the argument constructed is an unsigned integer.
-
-When the value of the
-additional information is less than 24, it is directly used as the
-argument's value.  When it is 24 to 27, the argument's value is held
-in the following 1, 2, 4, or 8, respectively, bytes, in network byte order.
+For major types except for 7, the additional information's value
+describes how to load an argument:
+
+Less than 24:
+: The argument's value is the value of the additional information.
```

Comment:
Sounds the wrong way around, but is consistent with the below.  Hmm, how to word this.